yarn-project/simulator/src/public/fixtures/public_tx_simulation_tester.ts

```diff
@@ -44,30 +44,43 @@ const defaultConfig: PublicSimulatorConfig = PublicSimulatorConfig.from({
   collectStatistics: false,
 });
 
+/**
+ * Factory type for creating a MeasuredPublicTxSimulatorInterface.
+ */
+export type MeasuredSimulatorFactory = (
+  merkleTree: MerkleTreeWriteOperations,
+  contractsDB: PublicContractsDB,
+  globals: GlobalVariables,
+  metrics: TestExecutorMetrics,
+  config: PublicSimulatorConfig,
+) => MeasuredPublicTxSimulatorInterface;
+
 /**
  * A test class that extends the BaseAvmSimulationTester to enable real-app testing of the PublicTxSimulator.
  * It provides an interface for simulating one transaction at a time and maintains state between subsequent
  * transactions.
  */
 export class PublicTxSimulationTester extends BaseAvmSimulationTester {
   protected txCount: number = 0;
-  private simulator: MeasuredPublicTxSimulatorInterface;
+  private simulator?: MeasuredPublicTxSimulatorInterface;
```

Comment:
I could be missing something, but I can't see why this is now optional (it's always set in the constructor)?